scripts/quests/windurst/Making_Amens.lua

```diff
@@ -0,0 +1,112 @@
+-----------------------------------
+-- Making Amens!
+-----------------------------------
+-- !addquest 2 8
+-- Kuroido-Moido : !pos -111 -4 101 240
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/titles')
+require('scripts/globals/zone')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.MAKING_AMENS)
+
+quest.reward =
+{
+    fame     = 75,
+    fameArea = xi.quest.fame_area.WINDURST,
+    title    = xi.title.HAKKURU_RINKURUS_BENEFACTOR,
+    gil      = 6000,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:hasCompletedQuest(xi.quest.log_id.WINDURST, xi.quest.id.windurst.MAKING_AMENDS) and
+                player:getFameLevel(xi.quest.fame_area.WINDURST) >= 4 and
+                not player:needToZone()
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Kuroido-Moido'] = quest:progressEvent(280),
+
+            onEventFinish =
+            {
+                [280] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and not player:hasKeyItem(xi.ki.BROKEN_WAND)
+        end,
+
+        [xi.zone.GARLAIGE_CITADEL] =
+        {
+            onEventFinish =
+            {
+                -- Mashira
+                [11] = function(player, csid, option, npc)
```

Comment:
we can move all of Mashira's work into this framework.

Reply:
I kept this out because [Rubbish Day](https://ffxiclopedia.fandom.com/wiki/Rubbish_Day) quest (Not yet converted) has shared conditions.
Seems like the cleanest way to handle it for now.

Reply:
Best to separate them and put all of his framework in here if any way possible. It's incredibly poor for code readability if we don't tie it all together

Reply:
It doesn't make sense to put this in the quest file because the NPC is like a utility that simply exists in the game and both quests use it (In this case to receive a key item if you're on the quest).

This would have to be duplicated in both quest files (Which is bad) and potentially clash and cause issues, if it's even possible to combine the conditions for both quests.

In the way I chose to handle, the NPC controls its dialogs because they are shared, and the quest specific code goes into the quest file. It makes the most sense to me.

Reply:
The entire point of the quest framework is to remove all code from individual NPC luas and into one centralized location. Rather than hunting down several areas to debug / work on the quest in the future.

Reply:
Hello a month later. I have to agree with Abdiah here, the idea behind the quest framework is to have the code relevant to the step of the quest they're on within the quest's lua. It would be best to duplicate it for each quest even though it does feel bad.